### ISSUE TITLE
False positive in object_usage_linter() for infix operators in glue strings

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2069]
```r
lint(linters = object_usage_linter(), "
foo <- function() {
  `%++%` <- `+`
  glue('{x %++% y}')
}")
# <text>:3:3: warning: [object_usage_linter] local variable '%++%' assigned but may not be used
#   `%++%` <- `+`
#   ^~~~~~
```

The problem is we only look for `SYMBOL` or `SYMBOL_FUNCTION_CALL` in the parse tree of the glue expression:

https://github.com/r-lib/lintr/blob/2813b6dcd29edb06676481df9cf879fd2c11e76c/R/object_usage_linter.R#L225

#2068 already fixed this by using `all.vars()` instead.
